src/WinGetServer/WinMain.cpp

```diff
@@ -18,27 +27,139 @@ static void _releaseNotifier() noexcept
+    // (ML;;NW;;;HI) specifies a high mandatory integrity level (requires admin).
+    // (A;;GA;;;UserSID) specifies access only for the user with the user SID (i.e. self).
+    wil::unique_hlocal_security_descriptor securityDescriptor;
+    std::string securityDescriptorString = "S:(ML;;NW;;;HI)D:(A;;GA;;;" + userSID + ")";
+    RETURN_LAST_ERROR_IF(!ConvertStringSecurityDescriptorToSecurityDescriptorA(securityDescriptorString.c_str(), SDDL_REVISION_1, &securityDescriptor, nullptr));
+
+    status = RpcServerRegisterIf3(WinGetServerManualActivation_v1_0_s_ifspec, nullptr, nullptr, RPC_IF_ALLOW_LOCAL_ONLY | RPC_IF_AUTOLISTEN, RPC_C_LISTEN_MAX_CALLS_DEFAULT, 0, nullptr, securityDescriptor.get());
+    RETURN_HR_IF(HRESULT_FROM_WIN32(status), status != RPC_S_OK);
+
+    return S_OK;
+}
+
+_Must_inspect_result_
+_Ret_maybenull_ _Post_writable_byte_size_(size)
+void* __RPC_USER MIDL_user_allocate(_In_ size_t size)
+{
+    return malloc(size);
+}
+
+void __RPC_USER MIDL_user_free(_Pre_maybenull_ _Post_invalid_ void* ptr)
 {
-    winrt::init_apartment();
+    if (ptr)
+    {
+        free(ptr);
+    }
+}
+
+extern "C" HRESULT CreateInstance(
+    /* [in] */ GUID clsid,
+    /* [in] */ GUID iid,
+    /* [in] */ UINT32 flags,
+    /* [ref][out] */ UINT32 * pcbBuffer,
+    /* [size_is][size_is][ref][out] */ BYTE * *ppBuffer)
+{
+    UNREFERENCED_PARAMETER(flags);
+
+    RETURN_HR_IF_NULL(E_POINTER, pcbBuffer);
+    RETURN_HR_IF_NULL(E_POINTER, ppBuffer);
+
+    wil::com_ptr<IStream> stream;
+    RETURN_IF_FAILED(CreateStreamOnHGlobal(nullptr, TRUE, &stream));
+
+    wil::com_ptr<IUnknown> instance;
+    RETURN_IF_FAILED(WindowsPackageManagerServerCreateInstance(&clsid, &iid, reinterpret_cast<void**>(&instance)));
+
+    RETURN_IF_FAILED(CoMarshalInterface(stream.get(), iid, instance.get(), MSHCTX_LOCAL, nullptr, MSHLFLAGS_NORMAL));
+
+    ULARGE_INTEGER streamSize{};
+    RETURN_IF_FAILED(stream->Seek({}, STREAM_SEEK_CUR, &streamSize));
+    RETURN_HR_IF(E_NOT_SUFFICIENT_BUFFER, streamSize.QuadPart > std::numeric_limits<UINT32>::max());
+
+    UINT32 bufferSize = static_cast<UINT32>(streamSize.QuadPart);
+
+    struct DeleteWithMidlFree { void operator()(void* m) { MIDL_user_free(m); } };
+    std::unique_ptr<BYTE, DeleteWithMidlFree> buffer{ reinterpret_cast<BYTE*>(MIDL_user_allocate(bufferSize)) };
+
+    RETURN_IF_FAILED(stream->Seek({}, STREAM_SEEK_SET, nullptr));
+    ULONG bytesRead = 0;
+    RETURN_IF_FAILED(stream->Read(buffer.get(), bufferSize, &bytesRead));
+    RETURN_HR_IF(E_UNEXPECTED, bytesRead != bufferSize);
+
+    *pcbBuffer = bufferSize;
+    *ppBuffer = buffer.release();
+
+    return S_OK;
+}
+
+int __stdcall wWinMain(_In_ HINSTANCE, _In_opt_ HINSTANCE, _In_ LPWSTR cmdLine, _In_ int)
+{
+    wil::SetResultLoggingCallback(&WindowsPackageManagerServerWilResultCallback);
+
+    RETURN_IF_FAILED(CoInitializeEx(nullptr, COINIT_MULTITHREADED));

```

Comment:
>CoInitializeEx [](http://example.com/codeflow?start=21&length=14)

Just curious why is the change to CoIntialize from init_apartments?

Reply:
We removed the dependency on the CppWinRT package so this is the alternative for initializing the COM library